hpctestlib/system/fs/mnt_opts.py

```diff
@@ -0,0 +1,127 @@
+    #:   and the value should be a string with options.
+    #:   E.g., {'xfs: 'nosuid,logbsize=32k'}
+    #: :default: ``{}``
+    fs_ref_opts = variable(typ.Dict, value={}, loggable=True)
+
+    executable = 'stat'
+    tags = {'system', 'fs'}
+
+    @loggable
+    @property
+    def fs_mnt_point(self):
+        '''The file system mount point
+
+        :type: :class:`str`
+        '''
+        return self.filesystems[0]
+
+    @loggable
+    @property
+    def fs_type(self):
+        '''The file system type
+
+        :type: :class:`str`
+        '''
+        return self.filesystems[1]
+
+    @loggable
+    @property
+    def fs_mnt_opts(self):
+        '''The file system mount options
+
+        :type: :class:`str`
+        '''
+        return self.filesystems[2]
+
+    @run_after('init')
+    def set_fs_opts(self):
+        # skip the test if the filesystem is not supported by the test
+        self.skip_if(self.fs_type not in self.fs_ref_opts,
+                     msg=f'This test does not support filesystem '
+                         f'type {self.fs_type}')
+
+        self.ref_opts=self.explode_opts_str(self.fs_ref_opts[self.fs_type])
+        self.curr_opts=self.explode_opts_str(self.fs_mnt_opts)
+
+    @run_after('init')
+    def set_executable_opts(self):
+        self.executable_opts = [self.fs_mnt_point]
+
+    def explode_opts_str(self, opts_str):
+        result=dict()
+        for opt in opts_str.split(','):
+            if opt == '':
+                continue
+            opt_splitted = opt.split('=')
+            keystr = opt_splitted[0]
+            valstr = opt_splitted[1] if len(opt_splitted) > 1 else ''
+            result[keystr] = valstr
+        return result
+
+    @sanity_function
+    def assert_config(self):
+        skip_me = sn.extractall('No such file or directory', self.stderr)
+        self.skip_if(skip_me, msg=f'Skipping test because filesystem '
+                                  f'{self.fs_mnt_point} was not found')
+
+        return sn.all(sn.chain(
+            sn.map(lambda x: sn.assert_in(x, self.curr_opts,
+                                          msg=f'Option {x} not present for '
+                                              f'mount point '
+                                              f'{self.fs_mnt_point}, '
+                                              f'of type {self.fs_type}'),
+                   self.ref_opts),
+            sn.map(lambda x: sn.assert_eq(x[1], self.curr_opts[x[0]],
+                                          msg=f'Value {x[1]} not equal to '
+                                              f'{self.curr_opts[x[0]]} in '
+                                              f'the variable {x[0]} for the '
+                                              f'mount point '
+                                              f'{self.fs_mnt_point}, '
+                                              f'of type {self.fs_type}'),
+                   self.ref_opts.items()),
+        ))
```

Comment:
I liked the `map` version much more than the for loop I created in the first implementation. So, I kept the `map` because it straight forward. In practice, it is just a single-line implementation. The message is the one that is taking most of the space there. If the message didn't depend on the lambda values, I would have defined it outside of the sanity function. And it would be a single liner...
 
What do you think?